set.mm

```diff
@@ -776576,6 +776859,676 @@ separated by (open) neighborhoods.  See ~ sepnsepo for the equivalence
+        WTXDXFWEWFAYRYPAEXPWNWPISWGWHYFAWPWRERZYQYLYFWPWREXBXENYSAWTXDXFWIWFAYT
+        YQAEXPWPWRISWGWHXRDXPWNWPWRYBYCWJWKAYJYKAEXPWNWRISWMWHYFWNWREXBVINYSYDY
+        FVJTYEYFVNTVKVLVOWL $.
+    $}
+  $}
+
+  ${
+    $d U f p x y z $.  $d f ph y z $.
+    setcthin.c $e |- ( ph -> C = ( SetCat ` U ) ) $.
+    setcthin.u $e |- ( ph -> U e. V ) $.
+    setcthin.x $e |- ( ph -> A. x e. U E* p p e. x ) $.
+    $( A category of sets whose objects contain at most one element is thin.
+       (Contributed by Zhi Wang, 20-Sep-2024.) $)
+    setcthin $p |- ( ph -> C e. ThinCat ) $=
+      ( vy vz vf cfv eqid cv wcel wa wmo mobidv adantr csetc chom setcbas eqidd
+      cthinc co wf weq elequ2 wral simprr rspcdva mofmo simprl elsetchom mpbird
+      syl ccat setccat isthincd eqeltrd ) ACDUAMZUEGAJKDVBLVBUBMZAVBDEVBNZHUCAV
+      CUDAJOZDPZKOZDPZQZQZLOZVEVGVCUFPZLRVEVGVKUGZLRZVJFOZVGPZFRZVNVJVOBOPZFRZV
+      QBDVGBKUHVRVPFBKFUISAVSBDUJVIITAVFVHUKZULFVEVGLUMUQVJVLVMLVJVBDVKVCEVEVGV
+      DADEPZVIHTVCNAVFVHUNVTUOSUPAWAVBURPHVBDEVDUSUQUTVA $.
+  $}
+
+  ${
+    $d x y z $.
+    $( The category ` ( SetCat `` 2o ) ` is thin.  A special case of
+       ~ setcthin .  (Contributed by Zhi Wang, 20-Sep-2024.) $)
+    setc2othin $p |- ( SetCat ` 2o ) e. ThinCat $=
+      ( vx vz vy c2o csetc cfv cthinc wcel wtru cvv eqidd 2oex a1i wmo wral csn
+      cv c0 wceq wo cpr wex elpri 0ex sneq eqeq2d spcev orim2i mo0sn 3syl df2o2
+      biimpri eleq2s rgen setcthin mptru ) DEFZGHIAUQDJBIUQKDJHILMBQAQZHBNZADOI
+      USADUSURRRPZUAZDURVAHURRSZURUTSZTVBURCQZPZSZCUBZTZUSURRUTUCVCVGVBVFVCCRUD
+      VDRSVEUTURVDRUEUFUGUHUSVHBCURUIULUJUKUMUNMUOUP $.
+  $}
+
+
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Preordered sets as thin categories
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $c ProsetToCat $.
+
+  $( Class function defining preordered sets as categories. $)
+  cprstc $a class ProsetToCat $.
+
+  ${
+    $d C f x y z $.  $d H f $.  $d K k x y z $.  $d X f $.  $d Y f $.
+    $d ph f x y z $.
+    $( Definition of the function converting a preordered set to a category.
+       Justified by ~ prsthinc .
+
+       This definition is somewhat arbitrary.  Example 3.3(4.d) of [Adamek]
+       p. 24 demonstrates an alternate definition with pairwise disjoint
+       hom-sets.  The behavior of the function is defined entirely, up to
+       isomorphism, by ~ prstcnid , ~ prstchom , and ~ prstcthin .  Other
+       important properties include ~ prstcbas , ~ prstcleval , ~ prstcle ,
+       ~ prstcocval , ~ prstcoc , ~ prstchom2 , and ~ prstcprs .  Use those
+       instead.
+
+       Note that the defining property ~ prstchom is equivalent to ~ prstchom2
+       given ~ prstcthin .  See ~ thincn0eu for justification.
+
+       "ProsetToCat" was taken instead of "ProsetCat" because the latter might
+       mean the category of preordered sets (classes).  However, "ProsetToCat"
```

Comment:
Again, the latter would typically be denoted by **Proset**, but I agree that `ProsetToCat` is clearer (and not too long IMO).

Reply:
The few examples I can think about are `toInc` (converting a set to its powet set, with the subset order), `toHL` (converting a lattice of closed subspaces to a pre-Hilbert space), `toPoly1` (converting a coefficient function to a polynomial), `toMetSp` (converting a metric to a metric space)...

All of them start with `to`, but in this case `toCat` would be too vague/too wide, so specifying we're converting a `Proset` seems indeed necessary.

On the other hand I think we had a convention that _actions_ (like all these example) would start with a lower case, while set extractors (like `Base` for example) would start with an upper case. Should we use this convention here?

Reply:
> Again, the latter would typically be denoted by **Proset**, but I agree that `ProsetToCat` is clearer (and not too long IMO).

I think people **Proset** in literature for the category of prosets but we have `SetCat` and `CatCat` in Metamath so for consistency...

Reply:
> The few examples I can think about are `toInc` (converting a set to its powet set, with the subset order), `toHL` (converting a lattice of closed subspaces to a pre-Hilbert space), `toPoly1` (converting a coefficient function to a polynomial), `toMetSp` (converting a metric to a metric space)...
> 
> All of them start with `to`, but in this case `toCat` would be too vague/too wide, so specifying we're converting a `Proset` seems indeed necessary.
> 
> On the other hand I think we had a convention that _actions_ (like all these example) would start with a lower case, while set extractors (like `Base` for example) would start with an upper case. Should we use this convention here?

I think following conventions and keeping consistencies are quite important but couldn't come up with a better idea...

Also, I don't think `toInc` converts to a power set? https://us.metamath.org/mpeuni/df-ipo.html

By the way, my plan is to combine `toInc` with `ProsetToCat` to create a category of open subsets. Long way to go though...

Reply:
Maybe `toThinCat`? But I am currently working on monoid as a category then we need to come up a new name instead of the one (`MndToCat`) I am taking...

Reply:
> Also, I don't think toInc converts to a power set? 

You're right, it converts to a proset.

Reply:
Thanks for the suggestion. `prosetToCat` doesn't look nice to me cause `Proset` is there; and `toThinCat` causes further issues as I mentioned above. I think I will just keep it as `ProsetToCat` before anyone comes up with a better one. Anyway it shouldn't be too difficult to change the names in the future, if necessary, right?

Reply:
Yes, changing names is easy. And I also don't really have better alternatives.